turbo/stages/headerdownload/header_algos.go

```diff
@@ -525,7 +519,8 @@ func (hd *HeaderDownload) InsertHeader(hf FeedHeaderFunc, terminalTotalDifficult
 			hd.moveLinkToQueue(link, NoQueue)
 			delete(hd.links, link.hash)
 			hd.removeUpwards(link)
-			return true, false, 0, nil
+			log.Warn("[downloader] Rejected header marked as bad", "hash", link.hash, "height", link.blockHeight)
+			return true, false, 0, lastTime, nil
 		}
 		if hd.ignoreAboveHeight != nil && link.blockHeight > *hd.ignoreAboveHeight {
 			return false, true, 0, nil
```

Comment:
../../turbo/stages/headerdownload/header_algos.go:526:27: not enough return values
	have (bool, bool, number, nil)
	want (bool, bool, uint64, uint64, error)